File: samples/custom-tests/java/v2/retrofit/UserApiTest.java
package io.swagger.client.api;

import io.swagger.client.ApiClient;
import io.swagger.client.model.User;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import com.github.tomakehurst.wiremock.WireMockServer;
import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
import java.net.HttpURLConnection;
import org.junit.AfterClass;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static com.github.tomakehurst.wiremock.client.WireMock.*;

/**
 * API tests for UserApi
 */
public class UserApiTest {

    private UserApi api;

    private static WireMockServer wireMockServer;

    @Before
    public void setup() {
        wireMockServer = new WireMockServer(WireMockConfiguration.wireMockConfig().dynamicPort());
        wireMockServer.start();

        configureFor(wireMockServer.port());

        ApiClient apiClient = new ApiClient();
        apiClient.getAdapterBuilder().setEndpoint("http://localhost:" + wireMockServer.port());
        api = apiClient.createService(UserApi.class);


        stubFor(post(urlPathMatching("/user"))
                .willReturn(aResponse()
                        .withStatus(HttpURLConnection.HTTP_OK)));

        stubFor(post(urlPathMatching("/user/createWithArray"))
                .willReturn(aResponse()
                        .withStatus(HttpURLConnection.HTTP_OK)));

        stubFor(post(urlPathMatching("/user/createWithList"))
                .willReturn(aResponse()
                        .withStatus(HttpURLConnection.HTTP_OK)));

        stubFor(delete(urlPathMatching("/user/user1"))
                .willReturn(aResponse()
                        .withStatus(HttpURLConnection.HTTP_OK)));

        stubFor(get(urlPathMatching("/user/user1"))
                .willReturn(aResponse()
                        .withStatus(HttpURLConnection.HTTP_OK)
                        .withBody("{\"id\":10,\"username\":\"user1\"}")));
        
    }

    @AfterClass
    public static void tearDown() {
        wireMockServer.stop();
    }

    
    /**
     * Create user
     *
     * This can only be done by the logged in user.
     */
    @Test
    public void createUserTest() {
        User body = new User().id(10L).username("user1");
        api.createUser(body);

        verify(exactly(1), postRequestedFor(urlEqualTo("/user")));
    }
    
    /**
     * Creates list of users with given input array
     *
     * 
     */
    @Test
    public void createUsersWithArrayInputTest() {
        User user = new User().id(10L).username("user1");
        List<User> body = Arrays.asList(user);
        api.createUsersWithArrayInput(body);

        verify(exactly(1), postRequestedFor(urlEqualTo("/user/createWithArray")));
    }
    
    /**
     * Creates list of users with given input array
     *
     * 
     */
    @Test
    public void createUsersWithListInputTest() {
        User user = new User().id(10L).username("user1");
        List<User> body = Arrays.asList(user);
        api.createUsersWithListInput(body);

        verify(exactly(1), postRequestedFor(urlEqualTo("/user/createWithList")));
    }
    
    /**
     * Delete user
     *
     * This can only be done by the logged in user.
     */
    @Test
    public void deleteUserTest() {
        String username = "user1";
        api.deleteUser(username);

        verify(exactly(1), deleteRequestedFor(urlEqualTo("/user/user1")));
    }
    
    /**
     * Get user by user name
     *
     * 
     */
    @Test
    public void getUserByNameTest() {
        String username = "user1";
        User response = api.getUserByName(username);

        Assert.assertEquals(10L, response.getId().longValue());
        Assert.assertEquals("user1", response.getUsername());

        verify(exactly(1), getRequestedFor(urlEqualTo("/user/user1")));
    }
}

File: samples/client/petstore/kotlin-retrofit2-client/src/test/kotlin/io/swagger/client/apis/UserApiTest.kt
package io.swagger.client.apis

import io.swagger.client.infrastructure.ApiClient
import io.swagger.client.models.User
import org.junit.Assert
import org.junit.Before
import org.junit.Test

/**
 * API tests for UserApi
 */
class UserApiTest {

    private lateinit var api: UserApi

    @Before
    fun setup() {
        api = ApiClient().createService(UserApi::class.java)
    }

    /**
     * Create user
     *
     * This can only be done by the logged in user.
     */
    @Test
    fun createUserTest() {
        val body: User? = null
        // val response = api.createUser(body).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Creates list of users with given input array
     *
     * 
     */
    @Test
    fun createUsersWithArrayInputTest() {
        val body: kotlin.collections.List<User>? = null
        // val response = api.createUsersWithArrayInput(body).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Creates list of users with given input array
     *
     * 
     */
    @Test
    fun createUsersWithListInputTest() {
        val body: kotlin.collections.List<User>? = null
        // val response = api.createUsersWithListInput(body).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Delete user
     *
     * This can only be done by the logged in user.
     */
    @Test
    fun deleteUserTest() {
        val username: kotlin.String? = null
        // val response = api.deleteUser(username).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Get user by user name
     *
     * 
     */
    @Test
    fun getUserByNameTest() {
        val username: kotlin.String? = null
        // val response = api.getUserByName(username).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Logs user into the system
     *
     * 
     */
    @Test
    fun loginUserTest() {
        val username: kotlin.String? = null
        val password: kotlin.String? = null
        // val response = api.loginUser(username, password).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Logs out current logged in user session
     *
     * 
     */
    @Test
    fun logoutUserTest() {
        // val response = api.logoutUser().execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Updated user
     *
     * This can only be done by the logged in user.
     */
    @Test
    fun updateUserTest() {
        val username: kotlin.String? = null
        val body: User? = null
        // val response = api.updateUser(username, body).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }
}
